lib/user.rb

```diff
@@ -0,0 +1,54 @@
+require './lib/database_connect'
+
+class User 
+  attr_reader :id, :name, :username
+
+  def initialize(id:, name:, username:)
+    @id = id 
+    @name = name 
+    @username = username
+  end 
+
+  def self.add(name:,username:, email:, password:)
+    DatabaseConnection.query("INSERT INTO users (name,username,email,password) 
+                      VALUES ('#{name}','#{username}','#{email}','#{password}')")
+  end 
+
+  def self.unique_username?(username)
+    result = DatabaseConnection.query("SELECT * FROM users WHERE username = '#{username}'").map { |row| row }
```

Comment:
Dont think you need to use a `map` here. 
Also, boolean logic can be simplified to `result.isEmpty?`